hybrasyl/Utility.cs

```diff
@@ -280,6 +281,116 @@ static class LogLevels
+        private string _xmlFileDirectory => Path.Combine(_worldFileDirectory, "xml");
+
+        public string MapFileDirectory => Path.Combine(_mapFileDirectory);
+
+        #region ScriptsFolders
+        public string ScriptCastableDirectory => Path.Combine(_scriptsFileDirectory, "castable");
+
+        public string ScriptCommonDirectory => Path.Combine(_scriptsFileDirectory, "common");
+
+        public string ScriptItemDirectory => Path.Combine(_scriptsFileDirectory, "item");
+
+        public string ScriptNpcDirectory => Path.Combine(_scriptsFileDirectory, "npc");
+
+        public string ScriptReactorDirectory => Path.Combine(_scriptsFileDirectory, "reactor");
+
+        public string ScriptStartupDirectory => Path.Combine(_scriptsFileDirectory, "startup");
+        #endregion
+
+        #region XmlFolders
+        public string XmlCastableDirectory => Path.Combine(_xmlFileDirectory, "castables");
+
+        public string XmlStatusDirectory => Path.Combine(_xmlFileDirectory, "statuses");
+
+        public string XmlItemDirectory => Path.Combine(_xmlFileDirectory, "items");
+
+        public string XmlNationDirectory => Path.Combine(_xmlFileDirectory, "nations");
+
+        public string XmlMapDirectory => Path.Combine(_xmlFileDirectory, "maps");
+
+        public string XmlWorldMapDirectory => Path.Combine(_xmlFileDirectory, "worldmaps");
+
+        public string XmlCreatureDirectory => Path.Combine(_xmlFileDirectory, "creatures");
+
+        public string XmlSpawnGroupDirectory => Path.Combine(_xmlFileDirectory, "spawngroups");
+
+        public string XmlLootSetDirectory => Path.Combine(_xmlFileDirectory, "lootsets");
+
+        public string XmlItemVariantDirectory => Path.Combine(_xmlFileDirectory, "itemvariants");
+
+        public string XmlNpcsDirectory => Path.Combine(_xmlFileDirectory, "npcs");
+
+        public string XmlLocalizationDirectory => Path.Combine(_xmlFileDirectory, "localization");
+        #endregion
+
+        public string[] DataSubFolders()
+        {
+            string[] dataSubFolders = new string[]
+                {
+                    MapFileDirectory,
+
+                    ScriptCastableDirectory,
+                    ScriptCommonDirectory,
+                    ScriptItemDirectory,
+                    ScriptNpcDirectory,
+                    ScriptReactorDirectory,
+                    ScriptStartupDirectory,
+
+                    XmlCastableDirectory,
+                    XmlStatusDirectory,
+                    XmlItemDirectory,
+                    XmlItemVariantDirectory,
+                    XmlNationDirectory,
+                    XmlMapDirectory,
+                    XmlWorldMapDirectory,
+                    XmlCreatureDirectory,
+                    XmlSpawnGroupDirectory,
+                    XmlLootSetDirectory,
+                    XmlNpcsDirectory,
+                    XmlLocalizationDirectory
+                };
+            return dataSubFolders;
+        }

```

Comment:
These are probably better implemented as static properties with constant values, or, use reflection to return a list of `*Directory` properties.